src/warnings.cpp

```diff
@@ -28,6 +29,12 @@ void SetfLargeWorkInvalidChainFound(bool flag)
     fLargeWorkInvalidChainFound = flag;
 }
 
+void SetMedianTimeOffsetWarning()
+{
+    LOCK(g_warnings_mutex);
+    g_timeoffset_warning = true;
```

Comment:
Why don't you implement going out of this warning?

Reply:
This wasn't implemented before either. This can be improved in a follow up.